icinga-app/icinga2.cmake

```diff
@@ -1,29 +1,39 @@
 #!/bin/sh
 ICINGA2_BIN=@CMAKE_INSTALL_FULL_LIBDIR@/icinga2/sbin/icinga2
 
-if test "x`uname -s`" = "xLinux" -a "x$1" = "xconsole"; then
-  libedit_line=`ldd $ICINGA2_BIN 2>&1 | grep libedit`
-  if test $? -eq 0; then
-    libedit_path=`echo $libedit_line | cut -f3 -d' '`
-    if test -n "$libedit_path"; then
-      libdir=`dirname -- $libedit_path`
-      found=0
-      for libreadline_path in `ls -1r -- $libdir/libreadline.so.* 2>/dev/null`; do
-        found=1
-        break
-      done
-      if test $found -eq 0; then
-        libdir2=/`echo $libdir | cut -f3- -d/`
-        for libreadline_path in `ls -1r -- $libdir2/libreadline.so.* 2>/dev/null`; do
+if test "x`uname -s`" = "xLinux"; then
+  if test "x$1" = "xconsole"; then
+    libedit_line=`ldd $ICINGA2_BIN 2>&1 | grep libedit`
+    if test $? -eq 0; then
+      libedit_path=`echo $libedit_line | cut -f3 -d' '`
+      if test -n "$libedit_path"; then
+        libdir=`dirname -- $libedit_path`
+        found=0
+        for libreadline_path in `ls -1r -- $libdir/libreadline.so.* 2>/dev/null`; do
           found=1
           break
         done
-      fi
-      if test $found -gt 0; then
-        export LD_PRELOAD="$libreadline_path:$LD_PRELOAD"
+        if test $found -eq 0; then
+          libdir2=/`echo $libdir | cut -f3- -d/`
+          for libreadline_path in `ls -1r -- $libdir2/libreadline.so.* 2>/dev/null`; do
+            found=1
+            break
+          done
+        fi
+        if test $found -gt 0; then
+          LD_PRELOAD="$libreadline_path:$LD_PRELOAD"
+        fi
       fi
     fi
   fi
+
+  for lib in /usr/lib{,64,/{{i386,x86_64}-linux-gnu,arm-linux-gnueabihf}}/libjemalloc.so.{1,2}; do
+    if test -e "$lib"; then
+      LD_PRELOAD="${lib}:$LD_PRELOAD"
+    fi
+  done
```

Comment:
So this will try to LD_PRELOAD basically everything it finds, regardless of architecture and whatnot. Realistic scenario that will likely happen at some point: `LD_PRELOAD=/usr/lib/i386-linux-gnu/libjemalloc.so.2` with a 64 bit icinga2 binary. I haven't tested what happens if you try this, but this doesn't sound right.

Reply:
And now:

```
root@5d8cbdfac947:/# ICINGA2_BIN=/usr/lib/x86_64-linux-gnu/icinga2/sbin/icinga2
root@5d8cbdfac947:/# for lib in /usr/lib{,64,/{{i386,x86_64}-linux-gnu,arm-linux-gnueabihf}}/libjemalloc.so.{1,2}; do
> echo $lib
> if test -e "$lib"; then
> echo EXISTS
> if LD_PRELOAD="$lib" "$ICINGA2_BIN" --version 2>&1 1>/dev/null |grep -qFwe LD_PRELOAD
> then echo INVALID
> else echo VALID
> fi
> fi
> done
/usr/lib/libjemalloc.so.1
/usr/lib/libjemalloc.so.2
/usr/lib64/libjemalloc.so.1
/usr/lib64/libjemalloc.so.2
/usr/lib/i386-linux-gnu/libjemalloc.so.1
/usr/lib/i386-linux-gnu/libjemalloc.so.2
EXISTS
INVALID
/usr/lib/x86_64-linux-gnu/libjemalloc.so.1
/usr/lib/x86_64-linux-gnu/libjemalloc.so.2
EXISTS
VALID
/usr/lib/arm-linux-gnueabihf/libjemalloc.so.1
/usr/lib/arm-linux-gnueabihf/libjemalloc.so.2
root@5d8cbdfac947:/#
```